### ISSUE TITLE
API for exposed units.

### ISSUE THREAD

**Issue by swager** [synth-inference/synthdid#44]
The current API requires the user to pass in data in a specific order: First N0 units are unexposed (controls), and the remaining N - N0 weights are exposed (treated after T0). This requires the user to do extra pre-processing.

We should do the following instead: We have a boolean (or 0/1) argument `exposed` in all methods that the user can use as they wish to denote which units are exposed. So they could use `exposed = 1:n > N0` to emulate current behavior, but can also pass in rows in other order if that's more convenient.

**Comment by erikcs** [comment]
A tangential item is that many users will most likely have their data in the typical long panel data setup (canonical input for almost all Stata estimators)

```unit_name year treated  covariate1  covariate2 ...```

and would need help reshaping.

**Comment by davidahirshberg** [comment]
Yeah. data(..) will load data in as a data frame. I'm thinking we should
have a function to.panel(dataframe, unit.field time.field, Y.field) that
makes the square panel from that format. Basically what people use daply in
plyr for.

**Comment by erikcs** [comment]
> Yeah. data(..) will load data in as a data frame. I'm thinking we should have a function to.panel(dataframe, unit.field time.field, Y.field) that makes the square panel from that format. Basically what people use daply in plyr for.
> […](#)

Yes, that should be a solid well tested function (missing values, unbalanced panel, etc, and be done for covariates X too) which only uses base R, with for example only matrix indexing https://github.com/synth-inference/synthdid/blob/master/experiments/california_smoking/basque_setup.R could be done with 

```R
panel = basque[, c('gdpcap', 'regionname', 'year')]
panel$treated = panel$regionname == 'Basque Country (Pais Vasco)'
colnames(panel) = c("outcome", "unit", "year", "treated")
panel = panel[order(panel$treated, panel$unit, panel$year), ]
T = length(unique(panel$year))
N = length(unique(panel$unit))
Ymat = matrix(panel$outcome, N, T, byrow = TRUE,
              dimnames = list(unique(panel$unit), unique(panel$year)))
T0 = which(colnames(Ymat) == 1967)
Ymat=Ymat[,1:(T0+5)]
all(Ymat - Y == 0)
[1] TRUE
```
